src/PublishPort.jsx

```diff
@@ -8,9 +9,45 @@ import { Popover } from "@patternfly/react-core/dist/esm/components/Popover";
 import { MinusIcon, OutlinedQuestionCircleIcon } from '@patternfly/react-icons';
 import cockpit from 'cockpit';
 
+import { FormHelper } from "cockpit-components-form-helper.jsx";
+import * as utils from './util.js';
+
 const _ = cockpit.gettext;
 
-export const PublishPort = ({ id, item, onChange, idx, removeitem, itemCount }) =>
+export function validatePublishPort(portMapping, key, validationFailed) {
+    const delta = { ...validationFailed };
+
+    switch (key) {
+    case "IP":
+        if (utils.isEmpty(portMapping.IP))
+            delta.IP = _("Required");
+        break;
+    case "hostPort": {
+        if (utils.isEmpty(portMapping.hostPort))
+            delta.hostPort = _("Required");
+
+        const hostPort = parseInt(portMapping.hostPort);
+        if (hostPort < 1 || hostPort > 65536)
+            delta.hostPort = _("Must be between 1 to 65536");
+
+        break;
+    }
+    case "containerPort": {
+        if (utils.isEmpty(portMapping.containerPort))
+            delta.containerPort = _("Required");
+
+        const containerPort = parseInt(portMapping.containerPort);
+        if (containerPort < 1 || containerPort > 65536)
+            delta.containerPort = _("Must be between 1 to 65536");
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20230713-111536-64696d17-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)